content/anchor-frontend.md

```diff
@@ -0,0 +1,619 @@
+
+- `connection` - the `Connection` to the Solana cluster
+- `wallet` - the `Wallet` object
+- `opts` - optional parameter that specifies the confirmation options, using a default setting if one is not provided
+
+Once we’ve create the `Provider` object, we then set it as the default provider using `setProvider`.
+
+```tsx
+import { useAnchorWallet, useConnection } from "@solana/wallet-adapter-react"
+import { AnchorProvider, setProvider } from "@project-serum/anchor"
+
+const { connection } = useConnection()
+const wallet = useAnchorWallet()
+const provider = new AnchorProvider(connection, wallet, {})
+setProvider(provider)
+```
+
+### Program
+
+The last step is to create a `Program` object that requires three parameters:
+
+- `IDL` - representing the structure of a program
+- `programId` - the on-chain address of the program with the structure of the `IDL`
+- `Provider` - establishing the `Connection` to a cluster and a `Wallet` for signing
+
+The `Program` object creates a custom API  we can use to interact with a Solana program. This API is the one stop shop for all things related to communicating with on-chain programs. Among other things, one can send transactions, fetch deserialized accounts, decode instruction data, subscribe to account changes, and listen to events. You can learn more about the `Program` class [here](https://coral-xyz.github.io/anchor/ts/classes/Program.html#constructor).
+
+To create the `Program` object, first import `Program` and `Idl` from `@project-serum/anchor`. The `Idl` type is used when working with Typescript.
+
+```tsx
+import { Program, Idl, AnchorProvider, setProvider } from "@project-serum/anchor"
+```
+
+Next, specify the `programId` of the program. We have to explicitly state the `programId` since there can be multiple programs with the same `IDL` structure (i.e. if the same program is deployed multiple times using different addresses). When creating the `Program` object, the default `Provider` is used if one is not explicitly specified.
+
+```tsx
+const programId = new PublicKey("PROGRAM_ADDRESS")
+const program = new Program(idl as Idl, programId)
+```
+
+All together, the final setup looks something like this:
+
+```tsx
+import idl from "./idl.json"
+import { useAnchorWallet, useConnection } from "@solana/wallet-adapter-react"
+import { Program, Idl, AnchorProvider, setProvider } from "@project-serum/anchor"
+
+const { connection } = useConnection()
+const wallet = useAnchorWallet()
+
+const provider = new AnchorProvider(connection, wallet, {})
+setProvider(provider)
+
+const programId = new PublicKey("PROGRAM_ADDRESS")
+const program = new Program(idl as Idl, programId)
+```
+
+## Anchor `MethodsBuilder`
+
+Once the `Program` object is set up, we can use the Anchor `MethodsBuilder` to build transactions with instructions from our program. The `MethodsBuilder` uses the `IDL` to provides a simplified format building transaction to invoke program instructions.
```

Comment:
typo:
The `MethodsBuilder` uses the `IDL` to provide a simplified format for building transactions to invoke program instructions.